client/lib/explat/internals/log-error.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * WordPress dependencies
+ */
+// TODO: Replace module in browser SSR context
+import { getLogger } from 'calypso/server/lib/logger';
+
+// Using typescript to ensure we are being safe in SSR
+declare const window: undefined | ( Window & typeof globalThis );
+
+export const isDevelopmentMode = process.env.NODE_ENV === 'development';
+
+/**
+ * Log an error to the server.
+ * Works in SSR.
+ *
+ * Should never throw.
+ *
+ * @param error Error to save
+ */
+export const logError = ( error: Record< string, string > & { message: string } ): void => {
+	try {
+		if ( isDevelopmentMode ) {
+			// eslint-disable-next-line no-console
+			console.error( '[ExPlat]', error.message, error );
+		}
+
+		error[ 'exPlatClientImplementation' ] = 'calypso';
+
+		if ( typeof window === 'undefined' ) {
+			getLogger()( error, `[ExPlat] ${ error.message }` );
```

Comment:
This should be `getLogger().error({context:'ExPlat', err:error})`:
* Bunyan automatically uses `error.message` for the `msg` field in the output
* Given this generates a structured error log (i.e. in JSON format), adding fields (like `context:"ExPlat"`, but feel free to use whatever name makes sense) is the preferred way to add extra info, it makes it easier to filter those logs later.

Reply:
Nice!